### PR TITLE
Handle multi-resource YAMLs in github subscription

### PR DESCRIPTION
**Pull request by rokej**

### CODE REVIEW

**Comment by rokej** [comment]
https://github.com/open-cluster-management/backlog/issues/1247

File: pkg/subscriber/github/subscriber_item.go [hunk 1]
```diff
@@ -950,33 +959,46 @@ func (ghsi *SubscriberItem) sortResources(repoRoot string, resourcePath string)
...
+			t := kubeResource{}
+			err := yaml.Unmarshal(resources[0], &t)
+
+			if err != nil {
+				fmt.Println("Failed to unmarshal YAML file")
+				// Just ignore the YAML
+				return nil
+			}
+
+			if t.APIVersion != "" && t.Kind != "" {
```
Comment (ianzhang366) [inline pkg/subscriber/github/subscriber_item.go]: Hi Roke, 

I'm missing some context over here, why do we want to handle these kinds separately?

Comment (rokej) [reply pkg/subscriber/github/subscriber_item.go]: The input to our system is a YAML with multiple kube resources. The output of our system that the kube client API we use takes one kube resource object to apply. So within our system, we need to handle this type of YAML, separate all resources and feed one resource at a time to the kube client.

Comment (ianzhang366) [reply pkg/subscriber/github/subscriber_item.go]: Yes, I understand that. 

Maybe my question is not clear, I meat to ask, in this `if` block, we are sperate `ghsi.rbacFiles`, `ghsi.crdsAndNamespaceFiles` against`ghsi.otherFiles `. I mean, why do you need this kind separation.

Comment (rokej) [reply pkg/subscriber/github/subscriber_item.go]: When there are mixture of kube resource kinds, we need to ensure that certain kinds of resources are applied first. For example, without a namespace, applying resouces into that namespace will fail. Without RBAC resource, applying other resources depending on the RBAC will fail. Line 150 applies these in such order. This ordering might not be complete today. Using kustomize ensures that the resource kinds are applied in correct order.

Comment (ianzhang366) [reply pkg/subscriber/github/subscriber_item.go]: Nice, new knowledge gained, thanks.

I was thinking, we will just let the resource fail, and the second or later sync loop will pick it up and apply.

**Pull request merged**